platform/darwin/src/MGLFillExtrusionStyleLayer.h

```diff
@@ -0,0 +1,281 @@
+// This file is generated.
+// Edit platform/darwin/scripts/generate-style-code.js, then run `make style-code-darwin`.
+
+#import "MGLFoundation.h"
+#import "MGLStyleValue.h"
+#import "MGLVectorStyleLayer.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ Controls the translation reference point.
+
+ Values of this type are used in the `MGLFillExtrusionStyleLayer.fillExtrusionTranslateAnchor`
+ property.
+ */
+typedef NS_ENUM(NSUInteger, MGLFillExtrusionTranslateAnchor) {
+    /**
+     The fill extrusion is translated relative to the map.
+     */
+    MGLFillExtrusionTranslateAnchorMap,
+    /**
+     The fill extrusion is translated relative to the viewport.
+     */
+    MGLFillExtrusionTranslateAnchorViewport,
+};
+
+/**
+ An extruded (3D) polygon.
+
+ You can access an existing fill-extrusion style layer using the
```

Comment:
Yes, as a noun phrase, it should be unhyphenated in prose. ("Fill-extrusion style layer" would be more natural as shorthand for "fill and extrusion style layer", like "zoom-property function" as shorthand for "zoom-and-property function".) Unlike in JavaScript, the hyphenated form `fill-extrusion` will never appear in the developer's code.